src/html.rs

```diff
@@ -137,24 +140,28 @@ impl HtmlMsgParser {
                     let mimetype = mail.ctype.mimetype.parse::<Mime>()?;
                     if mimetype == mime::TEXT_HTML {
                         if let Ok(decoded_data) = mail.get_body() {
-                            self.html = decoded_data;
+                            if self.html.is_empty() {
```

Comment:
Maybe swap this check with `mail.get_body()`, so we don't decode the part if we don't want to use it anyway.

Reply:
yip, that's better. 

i'll do a commit, and i also remove the `return Ok(());` as this is returned below anyway.